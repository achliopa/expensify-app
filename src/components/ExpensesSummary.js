import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';


// MY CODE 

// export const ExpensesSummary = (props) => (
// 	<div>
// 		{props.expenses.length > 0 && 
// 			<h3>
// 				Viewing {
// 					props.expenses.length
// 				} expense(s) totalling {
// 					numeral(selectExpensesTotal(props.expenses) /100)
// 					.format('0,0.00')
// 				}&euro;
// 			</h3>
// 		}
// 	</div>
// );

// const mapStateToProps = (state) => {
//   return {
//       expenses: selectExpenses(state.expenses,state.filters)
//   };
// };

// export default connect(mapStateToProps)(ExpensesSummary);

// COURSE CODE

export const ExpensesSummary = (props) => {
	const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = `${numeral(props.expensesTotal/100).format('0,0.00')}`;
	return (
		<div>
			<h2> 
				Viewing {props.expenseCount} {expenseWord} totalling {formattedExpensesTotal}&euro;
			</h2>
		</div>
	);
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses,state.filters);
  return {
      expenseCount: visibleExpenses.length,
      expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);