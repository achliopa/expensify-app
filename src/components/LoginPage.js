import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

// export default class LoginPage extends React.Component {
	
// 	onLogin= () => {
//         this.props.history.push('/dashboard');
//     };

// 	render () {
// 		return (
// 		    <div>
// 		        <button onClick={this.onLogin}>Login</button>
// 		    </div>
// 		);
// 	}
// }

export const LoginPage = ({startLogin}) => (
	<div className="box-layout">
		<div className="box-layout__box">
			<h1 className="box_layout__title">Agileng Expensify</h1>
			<p>It's time to put your expenses under control.</p>
			<button className="button" onClick={startLogin}>Login with Google</button>
		</div>
	</div>
);

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);