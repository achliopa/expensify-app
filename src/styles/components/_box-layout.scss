.box-layout {
    background: url('/images/bg.jpg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
}

.box_layout__box {
    background: fade-out(white, .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-layout: center;
    width: 25rem;
}

.box_layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}