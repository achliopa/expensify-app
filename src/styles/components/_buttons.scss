.login-button {
    background: $blue;
    border: none;
    color: white;
    font-weight: 500;
    padding: $s-size;
}