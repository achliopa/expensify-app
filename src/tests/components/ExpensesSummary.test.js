import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});