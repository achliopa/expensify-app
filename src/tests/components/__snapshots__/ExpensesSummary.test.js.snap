// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with no expenses 1`] = `<div />`;

exports[`should render ExpensesSummary with expenses 1`] = `<div />`;
